tests/link/test_numba.py

```diff
@@ -293,6 +294,15 @@ def test_create_numba_signature(v, expected, force_scalar):
     assert res == expected
 
 
+@pytest.mark.parametrize(
+    "input, aesara_type, aesara_wrapper",
+    [(np.random.RandomState(1), RandomStateType, shared)],
+)
+def test_aesaratypes_box_unbox(input, aesara_type, aesara_wrapper):
+    res = function([], aesara_wrapper(input), mode=numba_mode)()
+    assert aesara_type.filter(res)
```

Comment:
Doesn't look like the actual states stay consistent though. (looks like Numba's internal state changes, probably due to some unrelated operations, even if we aren't doing anything with it in the user side of the code.)

Normally doing something like this works:

```python
import numba
import numpy as np
from numba import _helperlib
rng = np.random.RandomState(1)
ints, index = rng.get_state()[1:3]
ptr = _helperlib.rnd_get_np_state_ptr()
_helperlib.rnd_set_state(ptr, (index, [int(x) for x in ints]))
state_list = _helperlib.rnd_get_state(_helperlib.rnd_get_np_state_ptr())[1]
assert np.all(ints == state_list) #Works
```

Reply:
Nevermind, they're consistent now. 

I had the misconception that `RandomState(state_list)` is equivalent to `RandomState().set_state(("MT19937", state_list, pos))`, spoiler alert, it isn't. The former actually uses the array as seed to generate an entirely different state array.